### PR TITLE
fix #2538 (Legacy): Update experiment versions after launch

### LINKED ISSUE #2538: Update versions from Normandy API
It's possible that the versions of a delivery in Normandy can be updated after the delivery is published, in which case Experimenter does not reflect the currently targeted versions.  We should update the versions in Experimenter when we read the Normandy API to update other things (like the status, etc).

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-160)
┆Issue Number: EXP-160

### PR DESCRIPTION
**Pull request by tiftran**
Because

* After an experiment is launched and versions are changed, it's not reflected

This commit
* Add an additional task to check to ensure versions in experiment and normandy are the same,
  else it updates experimenter to reflect these changes, with a changelog added

### CODE REVIEW

File: app/experimenter/normandy/tasks.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
     Experiment.STATUS_LIVE: Experiment.STATUS_COMPLETE,
 }
 
+NAMESPACE_FILTER_TOTAL = 10000
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tasks.py:27]: Is this needed?

File: app/experimenter/normandy/tasks.py [hunk 2]
```diff
@@ -152,15 +154,43 @@ def set_is_paused_value_task(experiment_id, recipe_data):
...
-                bucket_sample["count"] / bucket_sample["total"] * 100
+
+        update_population_percent(experiment, recipe_data, filter_objects)
+        update_firefox_versions(experiment, recipe_data, filter_objects)
+
+
+def update_population_percent(experiment, recipe_data, filter_objects):
+
+    if "bucketSample" in filter_objects:
+        bucket_sample = filter_objects["bucketSample"]
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tasks.py:168]: .....walrus assignment? >:D

File: app/experimenter/normandy/tasks.py [hunk 3]
```diff
@@ -152,15 +154,43 @@ def set_is_paused_value_task(experiment_id, recipe_data):
...
+            bucket_sample["count"] / bucket_sample["total"] * 100
+        )
+        experiment.save()
+
+
+def update_firefox_versions(experiment, recipe_data, filter_objects):
+    changed_data = {}
+
+    if "version" in filter_objects:
+        versions = filter_objects["version"]
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tasks.py:179]: :=

File: app/experimenter/normandy/tasks.py [hunk 4]
```diff
@@ -152,15 +154,43 @@ def set_is_paused_value_task(experiment_id, recipe_data):
...
+        min_version = min(versions["versions"])
+        max_version = max(versions["versions"])
+        if experiment.firefox_min_version != min_version:
+            changed_data["firefox_min_version"] = min_version
+        if experiment.firefox_max_version != max(versions):
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tasks.py:184]: dupe of line 181?

File: app/experimenter/normandy/tests/test_tasks.py [hunk 5]
```diff
@@ -335,6 +335,39 @@ def test_live_rollout_updates_population_percent(self):
...
+            normandy_id=1234,
+            firefox_min_version="80.0",
+            firefox_max_version="80.0",
+        )
+
+        mock_response_data = {
+            "approved_revision": {
+                "enabled": True,
+                "filter_object": [
+                    {"type": "version", "versions": [79.0, 80.0, 81.0, 82.0]}
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tests/test_tasks.py:350]: Do versions come in from normandy as floats like this? or as strings?

**Review by jaredlockhart — APPROVED** [review]
Looks good, couple little fixups thnx @tiftran! 👍

File: app/experimenter/normandy/tasks.py (hunk 4 above)
Comment (jaredlockhart) [reply app/experimenter/normandy/tasks.py:184]: Actually how does this even work, if `versions` is a dict and the actual version list is in `versions["versions"]`?

File: app/experimenter/normandy/tests/test_tasks.py (hunk 5 above)
Comment (tiftran) [reply app/experimenter/normandy/tests/test_tasks.py]: ooh you're right versions are listed at integers from normandy

**Comment by jaredlockhart** [comment]
Commit formatting nit:

`fix #bugnum type(scope): title`

**Pull request merged**